
project external/gtest/
diff --git a/include/gtest/internal/gtest-param-util.h b/include/gtest/internal/gtest-param-util.h
index 5559ab4..0db5fef 100644
--- a/include/gtest/internal/gtest-param-util.h
+++ b/include/gtest/internal/gtest-param-util.h
@@ -34,6 +34,7 @@
 #ifndef GTEST_INCLUDE_GTEST_INTERNAL_GTEST_PARAM_UTIL_H_
 #define GTEST_INCLUDE_GTEST_INTERNAL_GTEST_PARAM_UTIL_H_
 
+#include <malloc.h>
 #include <iterator>
 #include <utility>
 #include <vector>

project frameworks/base/
diff --git a/Android.mk b/Android.mk
index 810c060..051b6ee 100644
--- a/Android.mk
+++ b/Android.mk
@@ -220,6 +220,9 @@ LOCAL_JAVA_RESOURCE_FILES += $(LOCAL_PATH)/preloaded-classes
 
 #LOCAL_JARJAR_RULES := $(LOCAL_PATH)/jarjar-rules.txt
 
+LOCAL_CFLAGS := -fpermissive
+LOCAL_CXXFLAGS := -fpermissive
+
 LOCAL_DX_FLAGS := --core-library
 
 include $(BUILD_JAVA_LIBRARY)
diff --git a/libs/utils/RefBase.cpp b/libs/utils/RefBase.cpp
index bb6c125..c40a0ec 100644
--- a/libs/utils/RefBase.cpp
+++ b/libs/utils/RefBase.cpp
@@ -524,7 +524,7 @@ void RefBase::weakref_type::printRefs() const
 
 void RefBase::weakref_type::trackMe(bool enable, bool retain)
 {
-    static_cast<const weakref_impl*>(this)->trackMe(enable, retain);
+  static_cast</*const*/ weakref_impl*>(this)->trackMe(enable, retain); //TODO
 }
 
 RefBase::weakref_type* RefBase::createWeak(const void* id) const

project frameworks/compile/pixelflinger2/
diff --git a/src/main/compiler.h b/src/main/compiler.h
index 07933db..da66a5e 100644
--- a/src/main/compiler.h
+++ b/src/main/compiler.h
@@ -46,7 +46,7 @@
 #include <stdio.h>
 #include <string.h>
 #if defined(__linux__) && defined(__i386__)
-#include <fpu_control.h>
+//#include <fpu_control.h>
 #endif
 #include <float.h>
 #include <stdarg.h>
