diff --git a/Android.mk b/Android.mk
index edb9eee..23a23bd 100644
--- a/Android.mk
+++ b/Android.mk
@@ -116,7 +116,11 @@ USB_LAYOUT := $(LOCAL_PATH)/usb_layout.conf
 USB_IMAGE := $(PRODUCT_OUT)/$(TARGET_PRODUCT)_usb.img
 $(USB_IMAGE): $(usb_tmp_img) $(USB_LAYOUT) $(PRODUCT_OUT)/grub/grub.bin | $(edit_mbr)
 	@echo ----- Making usb image ------
-	@sed 's|default 2|default 0|' $(PRODUCT_OUT)/grub/grub.bin > $@
+
+#	http://groups.google.com/group/android-x86/browse_thread/thread/afae951f1775d39/baa5480e08025950?lnk=gst&q=ICS+USB+first+grub+menu+patch#baa5480e08025950
+#	@sed 's|default 1|default 0|' $(PRODUCT_OUT)/grub/grub.bin > $@ #orig
+	@sed 's|default=1|default=0|' $(PRODUCT_OUT)/grub/grub.bin | sed 's|hd1,0|hd0,0|' | sed 's|hd0,2|hd0,0|' > $@
+
 	@$(edit_mbr) -l $(USB_LAYOUT) -i $@ usb_boot=$(usb_tmp_img)
 
 .PHONY: iso_img usb_img
